src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1340,6 +1340,10 @@ private static bool IsLessDerivedThanAny<TMember>(int index, TypeSymbol type, Ar
                 {
                     return true;
                 }
+                else if (currentType.IsExtension && currentType.ExtendedTypeIsOrDerivedFrom(type, TypeCompareKind.ConsiderEverything, useSiteInfo: ref useSiteInfo))
+                {
+                    return true;
+                }
                 else if (currentType.IsClassType() && type.IsClassType() && currentType.IsDerivedFrom(type, TypeCompareKind.ConsiderEverything, useSiteInfo: ref useSiteInfo))
```

Comment:
>else if (currentType.IsClassType() && type.IsClassType() && currentType.IsDerivedFrom(type, TypeCompareKind.ConsiderEverything, useSiteInfo: ref useSiteInfo)) [](http://example.com/codeflow?start=16&length=158)

Should we get to this condition for extension type?
It looks like the entire `if` could be refactored to to not check for conditions that we know to fail due to prior checks.